python/rikai/types/vision.py

```diff
@@ -184,3 +197,36 @@ def to_numpy(self) -> np.ndarray:
         """Convert this image into an :py:class:`numpy.ndarray`."""
         with self.to_pil() as pil_img:
             return np.asarray(pil_img)
+
+    def crop(
+        self, box: Union[Box2d, List[Box2d]], format: Optional[str] = None
+    ) -> Union[Image, List[Image]]:
+        """Crop image specified by the bounding boxes, and returns the cropped
+        images.
+
+        Support crop images in batch, to save I/O overhead to download
+        the original image.
+
+        Parameters
+        ----------
+        box : :py:class:`Box2d` or :py:class:`List[Box2d]`
+            The bounding box(es) to crop out of this image.
+        format : str, optional
+            The image format to save as
+
+        Returns
+        -------
+        :py:class:`Image` or a list of :py:class:`Image`
+        """
+        if isinstance(box, Box2d):
+            boxes = [box]
```

Comment:
nit: maybe just clearer to have 
```
with self.to_pil().crop(bbox) as patch:
    return Image.from_pil(patch)
```
instead of converting to list then having an extra check later?

Reply:
confused. does this suggest `Image.from_pil` to take in a list of patch or sth?

or just handle list and single item differently.